Submodule deps/wasm3 contains modified content
diff --git a/deps/wasm3/source/m3_api_wasi.c b/deps/wasm3/source/m3_api_wasi.c
index 8b6f2e3..4931b26 100644
--- a/deps/wasm3/source/m3_api_wasi.c
+++ b/deps/wasm3/source/m3_api_wasi.c
@@ -6,6 +6,7 @@
 //
 
 #define _POSIX_C_SOURCE 200809L
+#define _GNU_SOURCE
 
 #include "m3_api_wasi.h"
 
@@ -44,7 +45,8 @@
 #          include <Security/Security.h>
 #      endif
 #  else
-#      include <sys/random.h>
+#      include <sys/syscall.h>
+#      include <linux/random.h>
 #  endif
 #  define HAS_IOVEC
 #elif defined(_WIN32)
@@ -680,7 +682,7 @@ m3ApiRawFunction(m3_wasi_generic_random_get)
         retlen = getentropy(buf, reqlen) < 0 ? -1 : reqlen;
 #   endif
 #elif defined(__FreeBSD__) || defined(__linux__)
-        retlen = getrandom(buf, buf_len, 0);
+        retlen = syscall(SYS_getrandom, buf, buf_len, 0);
 #elif defined(_WIN32)
         if (RtlGenRandom(buf, buf_len) == TRUE) {
             m3ApiReturn(__WASI_ERRNO_SUCCESS);

 
